### PR TITLE
PLANET-4599 EN Form block - Add custom_donate_url option and rename h…

### PR DESCRIPTION
**Pull request by sagarsdeshmukh**
…ide DONATE button field label

[JIRA 4599](https://jira.greenpeace.org/browse/PLANET-4599)

### CODE REVIEW

File: classes/blocks/class-enform.php [hunk 1]
```diff
@@ -247,8 +251,13 @@ public function prepare_data( $attributes ): array {
 			$attributes['thankyou_url'] = 'http://' . $attributes['thankyou_url'];
 		} else {
 			$options                   = get_option( 'planet4_options' );
-			$attributes['donatelink']  = $options['donate_button'] ?? '#';
 			$attributes['donate_text'] = $options['donate_text'] ?? __( 'Donate', 'planet4-engagingnetworks' );
+			if ( isset( $attributes['custom_donate_url'] ) && $attributes['custom_donate_url'] ) {
+				// Check if url start with http/https or not.
+				$attributes['donatelink'] = ( 0 !== strpos( $attributes['custom_donate_url'], 'http' ) ) ? 'http://' . $attributes['custom_donate_url'] : $attributes['custom_donate_url'];
```
Comment (Inwerpsel) [inline classes/blocks/class-enform.php]: It's probably better to preprend `https://` so we can be sure that the visitor will end on a page that has SSL enabled. Any site processing donations should do this over https. Maybe you can also check with a regex for `http(s):\/\/`?

Comment (sagarsdeshmukh) [reply classes/blocks/class-enform.php]: I agree with you for prepending https for donation page URL. but using regex over `strpos` will be expensive choice (where as both do same thing).

Comment (Inwerpsel) [reply classes/blocks/class-enform.php]: They don't do the same thing. Matching on `http` alone will match unintended strings, while really we only want to match strings that start with `http://` or `https://`. You could do that without a regex by using strpos twice with a different argument, but probably the overhead of having 2 function calls would be greater than the performance difference between `strpos` and a regex.

Comment (Inwerpsel) [reply classes/blocks/class-enform.php]: But it's not that important so I'll merge it anyway :)

**Pull request merged**